diff --git a/scripts/Makefile.lib b/scripts/Makefile.lib
index ac1ac97..be4ac4c 100644
--- a/scripts/Makefile.lib
+++ b/scripts/Makefile.lib
@@ -108,15 +108,15 @@ __a_flags	=                          $(call flags,_a_flags)
 __cpp_flags     =                          $(call flags,_cpp_flags)
 endif
 
-c_flags        = -Wp,-MD,$(depfile) $(NOSTDINC_FLAGS) $(CPPFLAGS) \
+c_flags        = -MD -MF $(depfile) $(NOSTDINC_FLAGS) $(CPPFLAGS) \
 		 $(__c_flags) $(modkern_cflags) \
 		 $(basename_flags) $(modname_flags)
 
-a_flags        = -Wp,-MD,$(depfile) $(NOSTDINC_FLAGS) $(CPPFLAGS) \
+a_flags        = -MD -MF $(depfile) $(NOSTDINC_FLAGS) $(CPPFLAGS) \
 		 $(__c_flags) \
 		 $(__a_flags) $(modkern_aflags)
 
-cpp_flags      = -Wp,-MD,$(depfile) $(NOSTDINC_FLAGS) $(__cpp_flags)
+cpp_flags      = -MD -MF $(depfile) $(NOSTDINC_FLAGS) $(__cpp_flags)
 
 # Seems to be a wrong thing to do. LDFLAGS contains gcc's flags,
 # yet ld_flags is fed to ld.
diff --git a/scripts/trylink b/scripts/trylink
index 2255dee..e74e3ae 100755
--- a/scripts/trylink
+++ b/scripts/trylink
@@ -96,7 +96,7 @@ fi
 START_GROUP="-Wl,--start-group"
 END_GROUP="-Wl,--end-group"
 INFO_OPTS() {
-	echo "-Wl,--warn-common -Wl,-Map,$EXE.map -Wl,--verbose"
+	echo ""
 }
 
 # gold may not support --sort-common (yet)
